db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLDecimal.java

```diff
@@ -537,7 +537,12 @@ protected int typeCompare(DataValueDescriptor arg) throws StandardException
     public DataValueDescriptor cloneValue(boolean forceMaterialization)
 	{
 		try {
-			return new SQLDecimal(getBigDecimal(), precision, scale);
+		    SQLDecimal decimal = new SQLDecimal(getBigDecimal(), precision, scale);
+		    if (decimal.getPrecision() != precision)
```

Comment:
The default behavior when constructing a SQLDecimal is to set the precision and scale to the that of the BigDecimal literal we're building:
`	private void setCoreValue(BigDecimal theValue)
	{
		value = theValue;
		if (value !=null) {
			precision = value.precision();
			scale = value.scale();
		}`

However, there is a special case where precision and scale may not match the underlying BigDecimal.  In IndexRowToBaseRowOperation, SPLICE-1700 added code to write column default values into the index template row:

`            //update compactRow with the default value info
            if (defaultRow != null && defaultValueMap != null) {
                for (int i=defaultValueMap.anySetBit(); i>=0; i=defaultValueMap.anySetBit(i)) {
                    compactRow.setColumnValue(i+1, defaultRow.getColumn(i+1));
                }
            }`

If the default value has scale and precision less than that of the column definition, it causes overflow when inserting values that should fit.  For example, if column is defined as Dec(5,2), but the default value is 100.1, the template row gets a DVD in it with data type DEC(3,1).  If we then try to populate the template row with a larger value, say 1000.1, this causes overflow, even though it should fit in DEC(5,2).  So, because of SPLICE-1700, we have a need to set the precision and scale of the DVD to something other than the precision and scale of the literal value held in the DVD (done by introducing new method setColumnValue instead of using setColumn).  Therefore, cloneValue needs to preserve the different scale and precision if it has been modified from the original value.  Since this is special-purpose code, I don't think we want to modify how all SQLDecimals are constructed just to accomodate this case.

Reply:
I see, that makes sense, thanks!